core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/solution/cloner/gizmo/GizmoSolutionClonerImplementor.java

```diff
@@ -52,8 +52,11 @@ public class GizmoSolutionClonerImplementor {
             Object.class);
     private static final MethodDescriptor PUT_METHOD = MethodDescriptor.ofMethod(Map.class, "put", Object.class,
             Object.class, Object.class);
+    private static final String FALLBACK_CLONER = "fallbackCloner";
     public static final boolean DEBUG = false;
 
+    public static boolean IS_QUARKUS = false;
```

Comment:
Let's figure out a different way to pass this information. Static mutable fields are fundamentally wrong, and we've seen it numerous times already.

Do we need a Quarkus-specific subtype? So be it.
Quarkus-specific constructor? Sure.
Just not this, please.

In my opinion, the fact that there are ifs for particular situations screams for two distinct implementations with a shared supertype, or the lambda/strategy pattern.